src/core/OnExceptionSkipOrExitDispatcher.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+
+namespace CR.MessageDispatch.Core
+{
+    public class OnExceptionSkipOrExitDispatcher<TMessage> : IDispatcher<TMessage>
+    {
+        private readonly IDispatcher<TMessage> _dispatcher;
+        private readonly Func<TMessage, Exception, bool> _shouldExit;
+
+        public OnExceptionSkipOrExitDispatcher(IDispatcher<TMessage> dispatcher, Func<TMessage, Exception, bool> shouldExit)
+        {
+            _dispatcher = dispatcher;
+            _shouldExit = shouldExit;
+        }
+
+        public OnExceptionSkipOrExitDispatcher(IDispatcher<TMessage> dispatcher, bool alwaysExit)
```

Comment:
Slightly weird if I set alwaysExit to false ... does that mean sometimesExit? Maybe theres a better name like `exitOnException`?

Actually thinking about it, why would you even use this wrapper if you are setting that to false, it does nothing?